docker-images/build.sh

```diff
@@ -65,14 +66,61 @@ function build {
     do
         sha=${version_checksums[$kafka_version]}
         lib_directory=${version_libs[$kafka_version]}
+        
+        # If there is a file specified for this version of Kafka use that instead of the specified URL
+        if [ ${version_binary_files[$kafka_version]} ]
+        then 
+
+            filepath=${version_binary_files[$kafka_version]}
+            echo "Kafka binary file field was specified, using $filepath instead of URL"
+            
+            # Extract the filename from the filepath
+            OIFS=$IFS
+            IFS="/"
+            read -a fparr <<< $filepath
+            filename=${fparr[-1]}
+            IFS=$OIFS
+            
+        elif [ ${version_binary_urls[$kafka_version]} ]
+        then
+
+            url=${version_binary_urls[$kafka_version]}
+            echo "Downloading Kafka $kafka_version binaries from: $url"
+
+            # Extract the filename from the URL
+            OIFS=$IFS
+            IFS="/"
+            read -ra urlarr <<< "$url"
+            filename=${urlarr[-1]}
+            IFS=$OIFS
+            
+            filepath="$(mktemp -d)/$filename"
+
+            curl --output "$filepath" "$url"
+
+        fi
+            
+        # Do the checksum on the Kafka binaries
+        kafka_checksum="$sha $filepath"
```

Comment:
Can we rename `sha` → `expected_sha` to be clearer about the role it's playing here?